modules/markup/html.go

```diff
@@ -356,40 +356,23 @@ func (ctx *postProcessCtx) visitNode(node *html.Node, visitText bool) {
 				if attr.Key != "class" {
 					continue
 				}
-				// Got a class
-				from := 0
-
-				for idx := strings.Index(attr.Val[from:], "icon") + from; idx >= from; {
-					if (idx != 0 && attr.Val[idx-1] != ' ') ||
-						(len(attr.Val) > idx+4 && attr.Val[idx+4] != ' ') {
-						from = idx + 4
-						continue
+				classes := strings.Split(attr.Val, " ")
+				for i, class := range classes {
+					if class == "icon" {
+						classes[0], classes[i] = classes[i], classes[0]
+						attr.Val = strings.Join(classes, " ")
+
+						// Remove all children of icons
+						child := node.FirstChild
+						for child != nil {
+							node.RemoveChild(child)
+							child = child.NextSibling
```

Comment:
```suggestion
							child = node.FirstChild
```
I wonder if `child.NextSibling` remains valid after `RemoveChild`.